workflow/setup_expt.py

```diff
@@ -59,148 +50,82 @@ def _update_defaults(dict_in: dict) -> dict:
+            output_config = f'{inputs.expdir}/{inputs.pslot}/{cfg_file}'  # output file in EXPDIR
+            cfg_key = '.'.join(cfg_file.split('.')[1:])  # key to look for in yaml_dict
+            _data = host_plus_inputs_dict.copy()
+            if cfg_key in yaml_dict.keys():
+                _data = AttrDict(_data, **yaml_dict[cfg_key])
+            Jinja(input_template, _data).save(output_config)
+        else:  # copy the file as is
+            input_file = f'{inputs.configdir}/{file}'
+            output_config = f'{inputs.expdir}/{inputs.pslot}/{file}'
+            shutil.copy(input_file, output_config)
 
     return
 
 
-def edit_baseconfig(host, inputs, yaml_dict):
-    """
-    Parses and populates the templated `HOMEgfs/parm/config/<gfs|gefs|sfs>/config.base`
-    to `EXPDIR/pslot/config.base`
-    """
-
-    # Create base_dict which holds templated variables to be written to config.base
-    base_dict = {
-        "@HOMEgfs@": _top,
-        "@MACHINE@": host.machine.upper()}
+@logit(logger)
+def map_inputs_to_configs(inputs):
 
-    if inputs.start in ["warm"]:
-        is_warm_start = ".true."
-    elif inputs.start in ["cold"]:
-        is_warm_start = ".false."
-    else:
-        raise ValueError(f"Invalid start type: {inputs.start}")
+    warm_start_map = {'warm': '.true.', 'cold': '.false.'}
 
     # Construct a dictionary from user inputs
-    extend_dict = {
-        "@PSLOT@": inputs.pslot,
-        "@SDATE@": datetime_to_YMDH(inputs.idate),
-        "@EDATE@": datetime_to_YMDH(inputs.edate),
-        "@CASECTL@": f'C{inputs.resdetatmos}',
-        "@OCNRES@": f"{int(100.*inputs.resdetocean):03d}",
-        "@EXPDIR@": inputs.expdir,
-        "@COMROOT@": inputs.comroot,
-        "@EXP_WARM_START@": is_warm_start,
-        "@MODE@": inputs.mode,
-        "@INTERVAL_GFS@": inputs.interval,
-        "@SDATE_GFS@": datetime_to_YMDH(inputs.sdate_gfs),
-        "@APP@": inputs.app,
-        "@NMEM_ENS@": getattr(inputs, 'nens', 0)
-    }
-
-    if getattr(inputs, 'nens', 0) > 0:
-        extend_dict['@CASEENS@'] = f'C{inputs.resensatmos}'
+    try:
+        dict_out = AttrDict({
+            "PSLOT": inputs.pslot,
+            "SDATE": to_YMDH(inputs.idate),
+            "EDATE": to_YMDH(inputs.edate),
+            "CASE_CTL": f'C{inputs.resdetatmos}',
+            "OCNRES": f"{int(100.*inputs.resdetocean):03d}",
+            "EXPDIR": inputs.expdir,
+            "COMROOT": inputs.comroot,
+            "EXP_WARM_START": warm_start_map[inputs.start],
+            "MODE": inputs.mode,
+            "INTERVAL_GFS": inputs.interval,
+            "SDATE_GFS": to_YMDH(inputs.sdate_gfs),
+            "APP": inputs.app,
+            "NMEM_ENS": getattr(inputs, 'nens', 0),
+            "ICSDIR": inputs.icsdir,
+            "ACCOUNT": inputs.account,
+        })
+    except Exception as ee:
+        raise Exception("Error in constructing dictionary from user inputs, check inputs: ", ee)
```

Comment:
This exception should be raised from ee
```suggestion
        raise Exception("Error in constructing dictionary from user inputs, check inputs: ") from ee